lib/jsdom/living/traversal/TreeWalker-impl.js

```diff
@@ -0,0 +1,197 @@
+"use strict";
+
+const DOMException = require("../../web-idl/DOMException");
+const idlUtils = require("../generated/utils");
+
+//FIXME: Once NodeFilter is ported to IDL method, uncomment these.
+const FILTER_ACCEPT = 1; //NodeFilter.FILTER_ACCEPT;
+const FILTER_REJECT = 2; //NodeFilter.FILTER_REJECT;
+const FILTER_SKIP = 3; //NodeFilter.FILTER_SKIP;
+const FIRST = false;
+const LAST = true;
+const NEXT = false;
+const PREVIOUS = true;
+
+function isNull(o) {
+	return o === null || typeof o === "undefined";
+}
+
+class TreeWalkerImpl {
+	_filterNode(node) {
+		const n = node.nodeType - 1;
+		if (!((1 << n) & this.whatToShow)) return FILTER_SKIP;
+		const filter = this.filter;
+		if (isNull(filter)) return FILTER_ACCEPT;
+		let result;
+
+		if (typeof filter === "function") result = filter(idlUtils.wrapperForImpl(node));  //non-standard
```

Comment:
It wasn't apart of any spec I could find. But seems like all browsers support defining the filter this way.